### PR TITLE
Fix the lock ID to return the correct value

### PR DESCRIPTION
**Pull request by johscheuer**
# Description

Fixes: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/1795

## Type of change

*Please select one of the options below.*

- Bug fix (non-breaking change which fixes an issue)

## Discussion

This bug was introduced in the migration from `v1beta1` to `v1beta2`.

## Testing

e2e test will be running and I will add additional unit tests.

## Documentation

-

## Follow-up

-

### CODE REVIEW

File: e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go [hunk 1]
```diff
@@ -474,7 +472,9 @@ var _ = Describe("Operator HA Upgrades", Label("e2e", "pr"), func() {
 				fixtures.PodsSelector(operatorPrimarySatellitePods.Items),
 			}, "20")
 
-			upgradeAndVerify(fdbCluster, targetVersion)
+			Expect(fdbCluster.UpgradeCluster(targetVersion, false)).NotTo(HaveOccurred())
```
Comment (sbodagala) [inline e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go:475]: What is this change about? "UpgradeCluster()" and "checkVersion()" are still getting called, which is what "upgradeAndVerify()" is doing too?

File: api/v1beta2/foundationdbcluster_types.go [hunk 2]
```diff
@@ -1848,9 +1848,9 @@ func (cluster *FoundationDBCluster) GetLockDuration() time.Duration {
 }
 
 // GetLockID gets the identifier for this instance of the operator when taking
-// locks.
+// locks. This is the `ProcessGroupIDPrefix` defined for this cluster.
 func (cluster *FoundationDBCluster) GetLockID() string {
-	return cluster.Spec.LockOptions.LockKeyPrefix
+	return cluster.Spec.ProcessGroupIDPrefix
```
Comment (sbodagala) [inline api/v1beta2/foundationdbcluster_types.go:1853]: Does "LockKeyPrefix" need to be deleted (if not used at any other place)?

Comment (johscheuer) [reply api/v1beta2/foundationdbcluster_types.go:1853]: The `LockKeyPrefix` iss used a few lines above, see:

```go
// GetLockPrefix gets the prefix for the keys where we store locking
// information.
func (cluster *FoundationDBCluster) GetLockPrefix() string {
	if cluster.Spec.LockOptions.LockKeyPrefix != "" {
		return cluster.Spec.LockOptions.LockKeyPrefix
	}
	return "\xff\x02/org.foundationdb.kubernetes-operator"
}
```

File: e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go (hunk 1 above)
Comment (johscheuer) [reply e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go:475]: The difference here is that we are not waiting until all clusters are fully reconciled. Waiting until the cluster is fully reconciled can take a long time and result in the test suite exceeding the time limits.

**Review by johscheuer — COMMENTED** [review]
Failing test:

```text
Summarizing 1 Failure:
  [FAIL] Operator HA Upgrades upgrading a HA cluster with link that drops some packets [It] Upgrade from 7.1.37 to 7.3.15 with network link that drops some packets [e2e, pr]
  /codebuild/output/src2369918553/src/github.com/FoundationDB/fdb-kubernetes-operator/e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go:111

Ran 7 of 8 Specs in 5632.658 seconds
FAIL! -- 6 Passed | 1 Failed | 1 Pending | 0 Skipped
--- FAIL: TestOperatorHaUpgrade (5633.23s)
FAIL
FAIL	github.com/FoundationDB/fdb-kubernetes-operator/e2e/test_operator_ha_upgrades	5633.262s
FAIL
```

I believe this is one of the tests that might be flaky based on its nature.

File: e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go (hunk 1 above)
Comment (sbodagala) [reply e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go:476]: Maybe we should let a small set of tests wait for all clusters to fully reconcile (by giving them more time to run), in order to find any issues related to that.

**Review by sbodagala — APPROVED** [review]

**Review by johscheuer — COMMENTED** [review]
Test failures:

```text

  [FAILED] Timed out after 600.001s.
  Expected
      <bool>: false
  to be true
  In [It] at: /codebuild/output/src235462494/src/github.com/FoundationDB/fdb-kubernetes-operator/e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go:111 @ 09/08/23 15:30:06.459
------------------------------
[AfterSuite] 
/codebuild/output/src235462494/src/github.com/FoundationDB/fdb-kubernetes-operator/e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go:55
[AfterSuite] PASSED [0.000 seconds]
------------------------------
[ReportAfterSuite] Autogenerated ReportAfterSuite for --junit-report
autogenerated by Ginkgo
[ReportAfterSuite] PASSED [0.495 seconds]
------------------------------

Summarizing 1 Failure:
  [FAIL] Operator HA Upgrades when no remote processes are restarted [It] Upgrade from 7.1.37 to 7.3.15 [e2e, pr]
  /codebuild/output/src235462494/src/github.com/FoundationDB/fdb-kubernetes-operator/e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go:111
```

This test is already known to be flaky:https://github.com/FoundationDB/fdb-kubernetes-operator/issues/1725

The other test failure seems to be a timing issue, where the Pod took a long time to be deleted (and the process was reported as an incompatible connection).

File: e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go (hunk 1 above)
Comment (johscheuer) [reply e2e/test_operator_ha_upgrades/operator_ha_upgrade_test.go:476]: That sounds like a good idea. We can update some of the tests to do that (or we create a dedicated suite for that).

**Pull request merged**